### DISCUSSION TITLE
Is there a way to set default conditional content?

### DISCUSSION THREAD

**Discussion by anielsen001** [quarto-dev/quarto-cli#8363] [Q&A] [answered]
### Description

I'm writing a document which outputs both a webpage and revealjs. The webpage contains a lot more information than the slidedeck will and I would like to set an overall condition on the document of don't render to revealjs unless explicitly tagged as such so that I don't have to put a condition on everything.

**Comment by mcanouil** [comment]
I am not sure I following what you mean by "default conditional content".
The default is to display everything to every format. It's up to you to tell when not to.
Use `when-format` and/or `unless-format` to control what is displayed where.
You can combine the two: `when-format="html" unless-format="revealjs"`.

**Reply by cscheid** [reply]
> You can combine the two: `when-format="html" unless-format="revealjs"`.

How is this going to behave differently from `when-format="html"`?

**Reply by mcanouil** [reply]
"revealjs" is "html" ...

That's the way it is implemented because of aliases masking formats: https://quarto.org/docs/authoring/conditional.html#format-matching

Initially reported: https://github.com/quarto-dev/quarto-cli/discussions/3215#discussioncomment-4095656

You have to do cumbersome things like this.

````qmd
---
format:
  revealjs: default
  html: default
---

::: {.content-visible when-format="html" unless-format="revealjs"}
# A title for HTML output
:::

::: {.content-visible when-format="revealjs"}
## A Slide for Reveal.js output
:::
````

**Reply by cscheid** [reply]
You're totally right, I forgot. That's indeed very annoying.

**Reply by mcanouil** [reply]
I forgot I actually made a PR to add this case to the documentation 😅
![image](https://github.com/quarto-dev/quarto-cli/assets/8896044/b6ae4fe6-dd5d-4d1b-9b13-1eb5eb4a0438)

**Comment by anielsen001** [comment]
What I mean is a default that is "nothing to revealjs unless specifically tagged". If I put reveajs as an output in the yaml frontmatter, it will all go to revealjs format unless I specifically tag it not to. I'm basically trying to avoid putting `{.content-visible when-format="html" unless-format="revealjs"}` all the time in the markdown and would like it all to be interpreted this way unless specified otherwise.

**Reply by mcanouil** [reply, marked as answer]
There is no other way then.

**Comment by anielsen001** [comment]
thank you
